### PR TITLE
Enable configuring the CA Certificate for UAA

### PR DESCRIPTION
**Pull request by reasonerjt**
Enable configuring the path of root cert of UAA in harbor.cfg.  It only
takes effects if the verify_cert is set to "true" If the file does not
exist, the configuration is skipped.
The intention for this commit is to support integration with nested UAA
in PAS or PKS, we don't expect user to manually configure this value,
though he can do it if he wants.

### CODE REVIEW

File: src/ui/auth/uaa/uaa.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ func CreateClient() (uaa.Client, error) {
 		ClientSecret:  UAASettings.ClientSecret,
 		Endpoint:      UAASettings.Endpoint,
 		SkipTLSVerify: !UAASettings.VerifyCert,
+		CARootPath:    os.Getenv("UAA_CA_ROOT"),
```
Comment (jessehu) [inline src/ui/auth/uaa/uaa.go]: Root CA is used commonly, so better to name it UAA_ROOT_CA?

File: make/prepare [hunk 2]
```diff
@@ -387,6 +389,16 @@ shutil.copyfile(os.path.join(templates_dir, "jobservice", "app.conf"), jobservic
...
 
+if auth_mode == "uaa_auth":
+    if os.path.isfile(uaa_ca_cert):
+        if not os.path.isdir(ui_cert_dir):
+            os.makedirs(ui_cert_dir, mode=0o600)
+        ui_uaa_ca = os.path.join(ui_cert_dir, "uaa_ca.pem")
+        print("Copying UAA CA cert to %s" % ui_uaa_ca)
+        shutil.copyfile(uaa_ca_cert, ui_uaa_ca)
+    else:
+        print("Can not find UAA CA cert: %s, skip" % uaa_ca_cert)
```
Comment (jessehu) [inline make/prepare]: should this be an error msg or warning msg ?

Comment (reasonerjt) [reply make/prepare]: warning, given the limitation of current approach of reading harbor.cfg

File: src/ui/auth/uaa/uaa.go (hunk 1 above)
Comment (reasonerjt) [reply src/ui/auth/uaa/uaa.go]: I think it's fine as it's an attribute in uaa.ClientConfig

**Pull request merged**